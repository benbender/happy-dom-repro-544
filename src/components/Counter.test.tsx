import userEvent from "@testing-library/user-event";
import { fireEvent, render, screen } from "solid-testing-library";
import { describe, expect, it } from "vitest";
import Counter  from "./Counter";

describe("<Counter />", () => {
	it("should render", () => {
		expect(Counter).toBeTruthy();

		const { unmount, container } = render(() => <Counter />);
		expect(container).toMatchSnapshot();

		unmount();
	});

	it("should increments value", async () => {
		expect(Counter).toBeTruthy();

		const { unmount } = render(() => <Counter />);
		const button = screen.queryByRole("button") as HTMLButtonElement;

		expect(button).toBeInTheDocument();
		expect(button).toHaveTextContent(/Clicks: 0/);

		fireEvent.click(button);
		expect(button).toHaveTextContent(/Clicks: 1/);

		await userEvent.click(button);
		expect(button).toHaveTextContent(/Clicks: 2/);

		unmount();
	});
});
