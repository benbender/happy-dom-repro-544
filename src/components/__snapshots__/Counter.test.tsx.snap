// Vitest Snapshot v1

exports[`<Counter /> > should render 1`] = `
<div>
  <button
    class="increment"
  >
    Clicks: 
    <!--#-->
    0
    <!--/-->
  </button>
</div>
`;
